cmd/clusterctl/client/cluster/mover.go

```diff
@@ -637,24 +636,23 @@ func pauseClusterClass(proxy Proxy, n *node, pause bool) error {
 	}
 
 	// Update the annotation to the desired state
-	ccAnnotations := clusterClass.GetAnnotations()
-	if ccAnnotations == nil {
-		ccAnnotations = make(map[string]string)
+	currentAnnotations := clusterClass.GetAnnotations()
+
+	// Create a copy of the current annotations to avoid modifying the original map.
```

Comment:
nit: Hm now that we are letting the patchhelper do the diffing we can also just modify the map :)
WDYT?